lib/modules/manager/npm/extract/common/package-file.ts

```diff
@@ -147,3 +147,24 @@ export function extractPackageJson(
     },
   };
 }
+
+export function hasPackageManager(content: string): boolean {
+  let packageJson: NpmPackage;
+  try {
+    logger.trace(`npm.hasPackageManager from package.json`);
+
+    packageJson = JSON.parse(content);
+    if (packageJson.packageManager) {
+      const match = regEx('^(?<name>.+)@(?<range>.+)$').exec(
+        packageJson.packageManager,
+      );
+
+      if (match?.groups) {
+        return true;
+      }
```

Comment:
```suggestion
      return regEx('^(?<name>.+)@(?<range>.+)$').test(
        packageJson.packageManager,
      );
```

simplify, i don't thin the regex test is required. just test for non-empty string. otherwise it will fail on other steps anyways.

Reply:
do now the check with the parsed result of the schema and check the transformed values